src/almanack/metrics/data.py

```diff
@@ -1182,3 +1195,71 @@ def find_doi_citation_data(repo: pygit2.Repository) -> Dict[str, Any]:
                 LOGGER.warning(f"Error during OpenAlex exact DOI lookup: {e}")
 
     return result
+
+
+def compute_sustainability_score(
+    table: List[Dict[str, Union[int, float, bool]]]
+) -> float:
+    """
+    Computes a sustainability score by normalizing
+    numeric metrics and incorporating boolean metrics.
+    Adjusts for metrics that should be inversely
+    proportional to the sustainability score.
+
+    Args:
+        data (List[Dict[str, Union[int, float, bool]]]):
+            A list of dictionaries containing metrics.
+            Each dictionary must have a "result" key
+            with a value that is an int, float, or bool.
+              Optionally, a "direction" key may be
+            included for numeric values to specify the
+            relationship to sustainability:
+            - 1 (positive correlation)
+            - 0 (no correlation)
+            - -1 (negative correlation)
+
+    Returns:
+        float:
+            The computed sustainability score, normalized between 0 and 1.
+    """
+    results = []
+    bool_results = []
+
+    # Gather numeric and boolean values with a direction
+    for item in table:
+        if isinstance(item["result"], bool) and item["direction"] != 0:
+            # for direction == 1 we treat True as positive correction
+            # and False as a negative correlation.
+            if item["direction"] == 1:
+                bool_results.append(1 if item["result"] else 0)
+            # for direction == -1 we treat True as negative correction
+            # and False as a positive correlation.
+            elif item["direction"] == -1:
+                bool_results.append(0 if item["result"] else 1)
+
+        elif isinstance(item["result"], (int, float)) and item["direction"] != 0:
+            # Numeric values inverted or kept as-is based on direction
+            results.append(
+                item["result"] if item["direction"] == 1 else 1 - item["result"]
+            )
+
+    # Normalize numeric values if any
+    if results:
+        scaler = RobustScaler()
+        normalized_results = scaler.fit_transform(
+            np.array(results).reshape(-1, 1)
+        ).flatten()
+
+        # Rescale to [0, 1] range
+        numeric_min = normalized_results.min()
+        numeric_max = normalized_results.max()
+        if numeric_max > numeric_min:  # Avoid division by zero
+            normalized_results = (normalized_results - numeric_min) / (
+                numeric_max - numeric_min
+            )
+    else:
+        normalized_results = []
+
+    # Combine normalized numeric and boolean results
+    all_results = list(normalized_results) + bool_results
+    return sum(all_results) / len(all_results) if all_results else 0
```

Comment:
Same comment about a placeholder zero vs. a real zero. Also, when would this compute to zero? I'm worried about silent errors and no exceptions here which would make us even more confused about placeholder zeros.